src/compiler/checker.ts

```diff
@@ -28948,10 +28948,13 @@ m2: ${(this.mapper2 as unknown as DebugTypeMapper).__debugToString().split("\n")
         }
 
         function reportObjectPossiblyNullOrUndefinedError(node: Node, facts: TypeFacts) {
+            const source = getSourceFileOfNode(node);
+            const text = getSourceTextOfNodeFromSourceFile(source, node);

```

Comment:
as https://github.com/microsoft/TypeScript/issues/8274#issuecomment-214026537 points out, we can't print arbitrary text in an error message because it might be too large.

We should use the original error message unless node is a EntityNameExpression (or EntityName? I can't remember the exact name) and text is less than some threshold. Maybe 100 characters?

Reply:
> it might be too large.

and span multiple lines too

Reply:
There is `isEntityNameExpression` and `entityNameToString` to give something more precise and useful.

https://github.com/microsoft/TypeScript/blob/94bb9500087d99aeeeb3c24e8d9c628ac523cffd/src/compiler/utilities.ts#L5112-L5114

https://github.com/microsoft/TypeScript/blob/94bb9500087d99aeeeb3c24e8d9c628ac523cffd/src/compiler/utilities.ts#L1016